File: app-android/app/src/main/java/de/tutao/tutanota/push/LocalNotificationInfo.java
package de.tutao.tutanota.push;


final class LocalNotificationInfo {
	final String message;
	final int counter;
	final PushMessage.NotificationInfo notificationInfo;

	LocalNotificationInfo(String message, int counter, PushMessage.NotificationInfo notificationInfo) {
		this.message = message;
		this.counter = counter;
		this.notificationInfo = notificationInfo;
	}

	LocalNotificationInfo incremented(int by) {
		return new LocalNotificationInfo(message, counter + by, notificationInfo);
	}

	@Override
	public String toString() {
		return "LocalNotificationInfo{" +
				"message='" + message + '\'' +
				", counter=" + counter +
				", notificationInfo=" + notificationInfo +
				'}';
	}
}

File: app-android/app/src/main/java/de/tutao/tutanota/push/LocalNotificationInfo.kt
package de.tutao.tutanota.push

data class LocalNotificationInfo(
		val message: String,
		val counter: Int,
		val notificationInfo: NotificationInfo
) {
	fun incremented(by: Int): LocalNotificationInfo {
		return LocalNotificationInfo(message, counter + by, notificationInfo)
	}
}